privval/signer_listener_endpoint.go

```diff
@@ -34,11 +34,13 @@ type SignerListenerEndpoint struct {
 	connectRequestCh      chan struct{}
 	connectionAvailableCh chan net.Conn
 
-	timeoutAccept time.Duration
-	pingTimer     *time.Ticker
-	pingInterval  time.Duration
+	timeoutAccept   time.Duration
+	acceptFailCount int
+	pingTimer       *time.Ticker
+	pingInterval    time.Duration
 
 	instanceMtx cmtsync.Mutex // Ensures instance public methods access, i.e. SendRequest
+	ivarMtx     cmtsync.Mutex // Ensures instance variable access, i.e. acceptFailCount
```

Comment:
```suggestion
        acceptFailCount int
	ivarMtx                cmtsync.Mutex // Ensures instance variable access, i.e. acceptFailCount
```

Reply:
It is probably enough to have an atomic integer here, but works the same way with that.

Reply:
I'm in favor of atomic int as well

Reply:
changed to atomic.Uint32